packages/react/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,114 @@
+import { Slot } from '@radix-ui/react-slot';
+import cl from 'clsx/lite';
+import {
+  Fragment,
+  type HTMLAttributes,
+  type ReactNode,
+  forwardRef,
+  useMemo,
+} from 'react';
+
+export type AvatarProps = {
+  /**
+   * The name of the person the avatar represents.
+   * Will be used to generate initials if no children are provided.
+   */
+  name?: string;
+  /**
+   * The color of the avatar.
+   *
+   * @default 'accent'
+   */
+  color?: 'accent' | 'neutral' | 'brand1' | 'brand2' | 'brand3';
+  /**
+   * The size of the avatar.
+   *
+   * @default 'md'
+   */
+  size?: 'xs' | 'sm' | 'md' | 'lg';
+  /**
+   * The shape of the avatar.
+   *
+   * @default 'circle'
+   */
+  variant?: 'circle' | 'square';
+  /**
+   * Image or icon to display inside the avatar.
+   *
+   * Gets `aria-hidden="true"`
+   */
+  children?: ReactNode;
+} & HTMLAttributes<HTMLSpanElement>;
+
+const fontSizeMap = {
+  xs: 'ds-paragraph--xs',
+  sm: 'ds-heading--2xs',
+  md: 'ds-heading--sm',
+  lg: 'ds-heading--md',
+};
+
+/**
+ * Avatars are used to represent people or entities.
+ *
+ * @example
+ * <Avatar name='John Doe' />
+ *
+ * @example
+ * <Avatar name='John Doe'>
+ *  <img src='...' alt='John Doe' />
+ * </Avatar>
+ *
+ * @example
+ * <Avatar name='John Doe'>
+ *  <Icon />
+ * </Avatar>
+ */
+export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(function Avatar(
+  {
+    name,
+    color = 'accent',
+    size = 'md',
+    variant = 'circle',
+    className,
+    children,
+    ...rest
+  },
+  ref,
+) {
+  const Component = children ? Slot : Fragment;
+
+  const initials = useMemo(() => {
+    return getInitials(name);
+  }, [name]);
+
+  return (
+    <span
+      ref={ref}
+      className={cl('ds-avatar', fontSizeMap[size], className)}
+      data-ds-variant={variant}
+      data-ds-color={color}
+      data-ds-size={size}
+      {...rest}
+      role='img'
+      aria-label={name}
+    >
+      <Component {...(children ? { 'aria-hidden': true } : {})}>
+        {children || initials}
+      </Component>
+    </span>
+  );
+});
+
+/**
+ * Gets initials using first and last word of a name.
+ */
+function getInitials(name: string | undefined): string | null {
+  if (!name) return null;
+  const initialis = [];
+  const segments = new Intl.Segmenter(document.documentElement.lang || 'no', {
+    granularity: 'word',
+  }).segment(name);
+  for (const segment of segments)
+    if (segment.isWordLike) initialis.push(segment.segment);
+  return `${initialis[0][0]}${initialis[initialis.length - 1][0]}`;
```

Comment:
```suggestion
  return `${initialis[0][0]}${initialis[initialis.length - 1][0] || ''}`;
```
To handle cases with just one name provided? :)

Reply:
Oh, should Ola become OO - are we design-wise sure about that? 🙈 
in that case we could just do `initialis.slice(-1)`

Reply:
No, we only want one initial when one word is provided. 
The code you suggested **would** give two initials in that case, since we know index 0 exists, meaning `|| ''` is redundant.
I changed it to this:
``${initialis[0][0]}${initialis.length > 1 ? initialis[initialis.length - 1][0] : ''}``